js/src/components/paid-ads/asset-group/asset-group-card.js

```diff
@@ -0,0 +1,236 @@
+
+	function refFirstErrorField( ref ) {
+		if ( firstErrorRef.current || getNumOfIssues( this ) === 0 ) {
+			return;
+		}
+		firstErrorRef.current = ref;
+	}
+
+	return (
+		<div key={ finalUrl } className="gla-asset-group-card">
+			{ ASSET_IMAGE_SPECS.map( ( spec ) => {
+				const initialImageUrls = baseAssetGroup[ spec.key ];
+				const imageProps = getInputProps( spec.key );
+
+				return (
+					<AssetField
+						key={ spec.key }
+						ref={ refFirstErrorField.bind( spec.key ) }
+						heading={ spec.heading }
+						subheading={ spec.subheading }
+						help={ spec.help }
+						numOfIssues={ getNumOfIssues( spec.key ) }
+						disabled={ ! isSelectedFinalUrl }
+						initialExpanded={ isSelectedFinalUrl }
+					>
+						<ImagesSelector
+							initialImageUrls={ initialImageUrls }
+							maxNumberOfImages={ spec.max }
+							imageConfig={ spec.imageConfig }
+							onChange={ imageProps.onChange }
+						>
+							{ renderErrors( spec.key ) }
+						</ImagesSelector>
+					</AssetField>
+				);
+			} ) }
+			{ ASSET_TEXT_SPECS.map( ( spec, index ) => {
+				const initialTexts = baseAssetGroup[ spec.key ];
+				const textProps = getInputProps( spec.key );
+
+				return (
+					<AssetField
+						key={ spec.key }
+						ref={ refFirstErrorField.bind( spec.key ) }
+						heading={ spec.heading }
+						subheading={
+							<>
+								{ spec.subheading }
+								{ index === 0 && isSelectedFinalUrl && (
+									<ExternalLink href="https://support.google.com/google-ads/answer/6167101">
+										{ __(
+											'Learn how to write effective ad copy',
+											'google-listings-and-ads'
+										) }
+									</ExternalLink>
+								) }
+							</>
+						}
+						help={ spec.help }
+						numOfIssues={ getNumOfIssues( spec.key ) }
+						disabled={ ! isSelectedFinalUrl }
+						initialExpanded={ isSelectedFinalUrl }
+					>
+						<TextsEditor
+							initialTexts={ initialTexts }
+							minNumberOfTexts={ spec.min }
+							maxNumberOfTexts={ spec.max }
+							maxCharacterCounts={ spec.maxCharacterCounts }
+							placeholder={ spec.capitalizedName }
+							addButtonText={ spec.addButtonText }
+							onChange={ textProps.onChange }
+						>
+							{ renderErrors( spec.key ) }
+						</TextsEditor>
+					</AssetField>
+				);
+			} ) }
+			<AssetField
+				className="gla-asset-field-call-to-action"
+				heading={ __( 'Call to action', 'google-listings-and-ads' ) }
+				help={ __(
+					'Select a call to action that aligns with your goals, or use automated call to action which allows Google to automatically choose the most relevant call to action for you.',
+					'google-listings-and-ads'
+				) }
+				disabled={ ! isSelectedFinalUrl }
+				initialExpanded={ isSelectedFinalUrl }
+			>
+				<SelectControl
+					options={ ctaOptions }
+					value={ ctaProps.value || ctaOptions[ 0 ].value }
+					onChange={ ctaProps.onChange }
+				/>
+			</AssetField>
+			<AssetField
+				ref={ refFirstErrorField.bind( 'display_url_path' ) }
```

Comment:
💅  WDYT of using a constant instead of hardcoding the string?

Reply:
Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/1863/commits/82af988567984bb248bba1d096000f96c30b6971.